pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/extension/PythonDefaults.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.extension;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import org.gradle.api.GradleException;
+
+
+public class PythonDefaults {
+    private final String defaultPython2Version;
+    private final String defaultPython3Version;
+    private final Collection<String> allowedVersions;
+
+    public PythonDefaults(String defaultPython2, String defaultPython3, Collection<String> allowedVersions) {
+        defaultPython2Version = defaultPython2;
+        defaultPython3Version = defaultPython3;
+        this.allowedVersions = allowedVersions;
+    }
+
+    public PythonDefaults(String defaultPython2, String defaultPython3) {
```

Comment:
I'm not sure what that means, can you provide more detail?

Here's my thinking: a site might want to say "we'll continue to allow the default set of acceptable Python versions, but we want the Python 2 default to be 2.7 and Python 3 to be 3.6".  In that case, it's convenient to have a constructor where they only have to set the defaults, not also the set of allowed versions.

The full constructor makes sense so they can set all parameters, and the empty constructor makes sense so that `PythonDetails` can just initialize itself with the built-in defaults.

One thing I thought about but did not implement, is ensuring that the default Python 2 and Python 3 versions appear in `allowedVersion`.  Without that, it would be possible to say "the default Python 3 version is 3.7" but to accidentally leave that out of the `allowedVersions` set.  I didn't make that change because it would certainly be user error to do that and it should be relatively easy to debug.  OTOH, it wouldn't be much code, so maybe it's a good idea to force the default Python versions into the allowed versions?  (It's a set after all, so can't do much harm.)